langchain/chains/query_constructor/query_language.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+from pydantic import BaseModel
+
+
+class Visitor(ABC):
```

Comment:
Could we make the docstring a bit more descriptive? I'm not really familiar with this cool stuff